### ISSUE TITLE
unstable kafka integration test generated columns

### ISSUE THREAD

**Issue by 3AceShowHand** [pingcap/tiflow#4939]
https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_kafka_integration_test/detail/cdc_ghpr_kafka_integration_test/9926/pipeline/151

**Comment by 3AceShowHand** [comment]
```
[2022/03/17 12:47:03.661 +08:00] [INFO] [main.go:546] ["DDL event received"] [DDL="{\"StartTs\":0,\"CommitTs\":431880240907419658,\"TableInfo\":{\"Schema\":\"generate_column\",\"Table\":\"\",\"TableID\":0,\"ColumnInfo\":null},\"PreTableInfo\":null,\"Query\":\"CREATE DATABASE `generate_column`\",\"Type\":1}"]
[2022/03/17 12:47:03.661 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880241182670862] [partition=3]
[2022/03/17 12:47:03.662 +08:00] [INFO] [main.go:546] ["DDL event received"] [DDL="{\"StartTs\":0,\"CommitTs\":431880241182670862,\"TableInfo\":{\"Schema\":\"generate_column\",\"Table\":\"t\",\"TableID\":0,\"ColumnInfo\":null},\"PreTableInfo\":null,\"Query\":\"CREATE TABLE `t` (`a` INT,`b` INT GENERATED ALWAYS AS(`a`+1) STORED PRIMARY KEY)\",\"Type\":3}"]
[2022/03/17 12:47:03.662 +08:00] [INFO] [main.go:546] ["DDL event received"] [DDL="{\"StartTs\":0,\"CommitTs\":431880242493652993,\"TableInfo\":{\"Schema\":\"generate_column\",\"Table\":\"t1\",\"TableID\":0,\"ColumnInfo\":null},\"PreTableInfo\":null,\"Query\":\"CREATE TABLE `t1` (`a` INT,`b` INT GENERATED ALWAYS AS(`a`+1) VIRTUAL NOT NULL,`c` INT NOT NULL,UNIQUE `idx1`(`b`),UNIQUE `idx2`(`c`))\",\"Type\":3}"]
[2022/03/17 12:47:03.662 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880242493652993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880246360276993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880246622420993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880246884564993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880247146708993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880247408590849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880247670734849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880247933140993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880248195022849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880248457428993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880248719572993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880248981716993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880249243598849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880249505742849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880249768148993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880250030030849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880250292174849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880250554318849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880250816724993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880251078606849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880251340750849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880251602894849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880251865038849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880252127182849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880252389588993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880252651470849] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880252913876993] [partition=3]
[2022/03/17 12:47:04.013 +08:00] [INFO] [main.go:506] ["update sink resolved ts"] [ts=431880253176020993] [partition=3]
[2022/03/17 12:47:03.662 +08:00] [PANIC] [main.go:533] ["DDL CommitTs < maxDDLReceivedTs"] [commitTs=431880240907419658] [maxDDLReceivedTs=431880242493652993] [DDL="{\"StartTs\":0,\"CommitTs\":431880240907419658,\"TableInfo\":{\"Schema\":\"generate_column\",\"Table\":\"\",\"TableID\":0,\"ColumnInfo\":null},\"PreTableInfo\":null,\"Query\":\"CREATE DATABASE `generate_column`\",\"Type\":1}"] [stack="main.(*Consumer).appendDDL\n\tgithub.com/pingcap/tiflow/cmd/kafka-consumer/main.go:533\nmain.(*Consumer).ConsumeClaim\n\tgithub.com/pingcap/tiflow/cmd/kafka-consumer/main.go:461\ngithub.com/Shopify/sarama.(*consumerGroupSession).consume\n\tgithub.com/Shopify/sarama@v1.27.2/consumer_group.go:690\ngithub.com/Shopify/sarama.newConsumerGroupSession.func2\n\tgithub.com/Shopify/sarama@v1.27.2/consumer_group.go:615"]
```
As you can see, DDL'CommitTs < maxDDLReceivedTs, this can happen when the DDL received from a lag partition.

**Comment by 3AceShowHand** [comment]
closed by #4938
